src/server/routes/attendees/create-attendee.ts

```diff
@@ -0,0 +1,43 @@
+import { Request, IReply } from 'hapi'
+import { MongoDBErrors } from '../../models'
+import { AttendeeModel } from '../../models'
+import { AttendeeResource } from '../../../resources'
+import * as Boom from '../../boom'
+
+export default async function handler(request: Request, reply: IReply) {
+  const requestDoc: AttendeeResource.TopLevelDocument = request.payload
+
+  if (!requestDoc
+    || !requestDoc.data
+    || !requestDoc.data.id
+    || typeof requestDoc.data.id !== 'string'
+    || !requestDoc.data.type
+    || requestDoc.data.type !== 'attendees') {
+    reply({}).code(400)
+    return
+  }
+
+  const attendee = new AttendeeModel({
+    attendeeid: requestDoc.data.id,
+  })
+
+  try {
+    await attendee.save()
+  } catch (err) {
+    if (err.code === MongoDBErrors.E11000_DUPLICATE_KEY) {
+      reply(Boom.conflict('Attendee already exists'))
+      return
+    }
+    throw err
+  }
+
+  const attendeeResponse = {
+    links: { self: `/attendees/${encodeURIComponent(attendee.attendeeid)}` },
+    data: {
+      type: 'attendees',
+      id: attendee.attendeeid,
+    },
+  } as AttendeeResource.TopLevelDocument
+
+  reply(attendeeResponse).code(201)
```

Comment:
It seems to be tested and passes, so I guess I must have done this in a plugin. I'll figure it out and make it common across the routes.